src/server/game/Movement/MotionMaster.cpp

```diff
@@ -88,9 +88,6 @@ void MotionMaster::UpdateMotion(uint32 diff)
     if (!_owner)
         return;
 
-    if (_owner->HasUnitState(UNIT_STATE_ROOT | UNIT_STATE_STUNNED)) // what about UNIT_STATE_DISTRACTED? Why is this not included?
-        return;
-
```

Comment:
it should be the other way around imo: if this condition is checked here, it should be removed from the MoveGens. This reduce code duplication.